src/event.rs

```diff
@@ -74,9 +74,93 @@ pub enum Event<'a, T: 'static> {
     UserEvent(T),
 
     /// Emitted when the application has been suspended.
+    ///
+    /// # Portability
+    ///
+    /// Not all platforms support the notion of suspending applications, and there may be no
+    /// technical way to guarantee being able to emit a `Suspended` event if the OS has
+    /// no formal application lifecycle (currently only Android and iOS do). For this reason
+    /// Winit does not currently try to emit pseudo `Suspended` events before the application
+    /// quits on platforms without an application lifecycle.
+    ///
+    /// In case of corner cases with how platform-specific events are mapped to Winit `Suspended`
+    /// and `Resumed` events it's recommended that applications be able to gracefully handle redundant
+    /// (i.e. back to back) `Suspended` or `Resumed` events.
+    ///
+    /// Also see [`Resumed`][Self::Resumed] notes.
+    ///
+    /// # Android
+    ///
+    /// On Android specifically, the `Suspended` event relates to two things:
+    /// 1. The application is paused (according to the [Activity lifecycle][android_ls])
+    /// 2. Or the application has had it's associated SurfaceView destroyed (which is expected
+    /// to closely correlate with being paused)
+    ///
+    /// Applications that need to run on Android should assume their SurfaceView has been
+    /// destroyed, which indirectly invalidates any existing render surfaces that may have been
+    /// created outside of Winit (such as an `EGLSurface`, `VkSurfaceKHR` or [`wgpu::Surface`][wgpu_surface]).
+    ///
+    /// [wgpu_surface]: https://docs.rs/wgpu/latest/wgpu/struct.Surface.html
```

Comment:
You can also make this `` [`wgpu::Surface`] `` and remove the `[wgpu_surface]` link target entirely. Same for `android_ls`.

Shall we also add links for `EGLSurface` and (Ash's) `VkSurfaceKHR`?

Reply:
yep the short hand wgpu::Surface linking makes sense.

There isn't really a canonical link for `EGLSurface` - the official EGL spec is a Khronos PDF. The Vulkan spec is at least published as HTML so could maybe link to the spec but not sure it's that important for either really.

Not sure about the "Same for `android_ls`" comment - that's a link about the Android activity life cycle so there wouldn't be a shorthand way of making that link?

Reply:
You can make that an `[Android lifecycle]: https...` too.

Yeah, link Vulkan if easy, I tried linking (E)GL in the ndk (see recent doc improvement PRs) and it's a bit of a hit and miss. Only need this if it's easy and makes it more clear to the user what you're talking about / referencing.